relnotes/array.feature.md

```diff
@@ -0,0 +1,8 @@
+### Assigns elements to a static or dynamic array.
+
+* `ocean.core.Array`
+
+  Add a new helper function `assignToArray()` to assign all the elements of an
```

Comment:
I'd like to see an explanation of why this is useful / in what situation it can help. It's not clear to me from reading this release note.

Reply:
The explanation is below, but I think it would be much clearer with a before/after example.

Reply:
Yeah, the way it's currently written, there's a lot of assumed knowledge.

Reply:
This is pretty much the reason in an example
```D
void foo ()
{
    int[4] arr = [0, 1, 2, 3];
}

void bar (int var0, int var1, int var2, int var3)
{
    int[4] array;
    array[0] = var0;
    array[1] = var1;
    array[2] = var2;
    array[3] = var3;
}

testNoAlloc(foo); // fail, it allocates memory in the heap

testNoAlloc(bar(1, 2, 3, 4)); // pass, but it does not scale well
```

Reply:
so it is helpful to have `array.fill(1,2,3,4)` and it won't allocate memory in the heap

Reply:
I see. Weird.

Would there ever be any reason to use this function on a dynamic array?

Reply:
I personally don't have any use case for a dynamic array but many for static ones

Reply:
Yeah, I think it's generally useful. If you want to fill the array with some values, you're in the same pickle as with static array. I'm not sure if we could have a version for static and for dynamic array separately, or use `static if` to differentiate the type of the destination array.

If so, we could even resize the dynamic array and call `assumeSafeAppend` inside it.

Reply:
What I was wondering is whether this would be more suited as a "create array" function that returns a static array of the appropriate size. (Though it'd need to use variadic _template_ args, in that case.)

Reply:
Full D2 transition will bring us stuff as https://github.com/dlang/phobos/blob/master/std/array.d#L4037-L4070

Reply:
Ah right, yeah, static arrays are only value types in D2 :)

Reply:
The real work is done here https://github.com/dlang/phobos/blob/master/std/array.d#L4136-L4180 but that wouldn't work in D1

Reply:
> If so, we could even resize the dynamic array and call assumeSafeAppend inside it.

I was trying to avoid going that way. My original idea was to ensure allocations will not happen